pkgs/development/tools/qtcreator/default.nix

```diff
@@ -1,84 +1,82 @@
+, cmake, qtbase, qt5compat, qtdeclarative, qtquick3d, qtquicktimeline
+, qtserialport, qtsvg, qttools, wrapQtAppsHook
+#, qtwebengine
+, llvmPackages, elfutils, rustc-demangle, perf, pkg-config
 , withDocumentation ? false, withClangPlugins ? true
 }:
 
 with lib;
 
-let
-  # Fetch clang from qt vendor, this contains submodules like this:
-  # clang<-clang-tools-extra<-clazy.
-  clang_qt_vendor = llvmPackages_8.clang-unwrapped.overrideAttrs (oldAttrs: {
-    # file RPATH_CHANGE could not write new RPATH
-    cmakeFlags = [ "-DCMAKE_SKIP_BUILD_RPATH=ON" ];
-    src = fetchgit {
-      url = "https://code.qt.io/clang/clang.git";
-      rev = "c12b012bb7465299490cf93c2ae90499a5c417d5";
-      sha256 = "0mgmnazgr19hnd03xcrv7d932j6dpz88nhhx008b0lv4bah9mqm0";
-    };
-    unpackPhase = "";
-  });
-in
-
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "qtcreator";
-  version = "5.0.3";
+  version = "8.0.1";
   baseVersion = builtins.concatStringsSep "." (lib.take 2 (builtins.splitVersion version));
-
   src = fetchurl {
-    url = "http://download.qt-project.org/official_releases/${pname}/${baseVersion}/${version}/qt-creator-opensource-src-${version}.tar.xz";
-    sha256 = "1sz21ijzvhf5avblikffykbqa8zdq3sbg32g2dmyxv5w211v3lvz";
+    url = "https://download.qt.io/official_releases/${pname}/${baseVersion}/${version}/qt-creator-opensource-src-${version}.tar.xz";
+    sha256 = "sha256-4s4gCnnHTc1jZ9y7g8g5wcILLMB31qZYY56s3opKuGU=";
   };
 
-  buildInputs = [ qtbase qtscript qtquickcontrols qtdeclarative elfutils.dev ] ++
-    optionals withClangPlugins [ llvmPackages_8.libclang
-                                 clang_qt_vendor
-                                 llvmPackages_8.llvm ];
-
-  nativeBuildInputs = [ qmake ];
-
-  # 0001-Fix-clang-libcpp-regexp.patch is for fixing regexp that is used to
-  # find clang libc++ library include paths. By default it's not covering paths
-  # like libc++-version, which is default name for libc++ folder in nixos.
-  # ./0002-Dont-remove-clang-header-paths.patch is for forcing qtcreator to not
-  # remove system clang include paths.
-  patches = [ ./0001-Fix-clang-libcpp-regexp.patch
-              ./0002-Dont-remove-clang-header-paths.patch ];
+  buildInputs = [
+      qtbase qt5compat qtdeclarative qtquick3d qtquicktimeline qtserialport
+      qtsvg qttools elfutils.dev rustc-demangle
+      #qtwebengine
+    ] ++
+    optionals withClangPlugins [
+      llvmPackages.libclang
+      llvmPackages.clang-unwrapped
+      llvmPackages.llvm
+    ];
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    wrapQtAppsHook
+  ];
 
   doCheck = true;
 
+  postPatch = ''
+    stat src/libs/extensionsystem/pluginmanager.cpp
+    cp ${./src/qt-creator-opensource-src-8.0.1/src/libs/extensionsystem/pluginmanager.cpp} src/libs/extensionsystem/pluginmanager.cpp
+    cp ${./src/qt-creator-opensource-src-8.0.1/src/plugins/welcome/welcomeplugin.cpp} src/plugins/welcome/welcomeplugin.cpp
```

Comment:
Can we fetch those files from somewhere instead?

Reply:
this is for debugging, will remove later